botocore/client.py

```diff
@@ -276,45 +276,28 @@ def _set_s3_presign_signature_version(self, client_meta,
         if provided_signature_version is not None:
             return
 
-        # Check to see if the region is a region that we know about. If we
-        # don't know about a region, then we can safely assume it's a new
-        # region that is sigv4 only, since all new S3 regions only allow sigv4.
-        regions = self._endpoint_resolver.get_available_endpoints(
-            's3', client_meta.partition)
-        if client_meta.region_name not in regions:
-            return
-
-        # If it is a region we know about, we want to default to sigv2, so here
-        # we check to see if it is available.
-        endpoint = self._endpoint_resolver.construct_endpoint(
-            's3', client_meta.region_name)
-        signature_versions = endpoint['signatureVersions']
-        if 's3' not in signature_versions:
-            return
-
-        # We now know that we're in a known region that supports sigv2 and
-        # the customer hasn't set a signature version so we default the
-        # signature version to sigv2.
+        # At this point, we want to default to sigV4 for all regions.
+        # SigV2 is no longer an option.
         client_meta.events.register(
-            'choose-signer.s3', self._default_s3_presign_to_sigv2)
+            'choose-signer.s3', self._default_s3_presign_to_sigv4)
 
-    def _default_s3_presign_to_sigv2(self, signature_version, **kwargs):
+    def _default_s3_presign_to_sigv4(self, signature_version, **kwargs):
```

Comment:
I'm not sure if we still need this handler? It may be interesting to see if we can just remove the `_set_s3_presign_signature_version` method and see if botocore just defaults to sigv4. We really only needed this logic because the presigners would default to sigv2 while the API requests would default to sigv4.